### PR TITLE
[Bugfix] add masking logits using vocab size to prevent OverflowError while detokenization

### PR DESCRIPTION
**Pull request by ashgold**
## Purpose
This PR is to fix the issue: https://github.com/vllm-project/vllm/issues/24211

This error occurs when the generated token ID is not present in the vocabulary.
To investigate why this issue occurs, I checked the lm_head dimension and vocabulary size of DeepSeek-R1.
In config.json, vocab_size is set to 129280,
but in tokenizer_config.json, the largest token ID is 118814.
I know the model was trained only on datasets with token IDs ≤ 118814, and IDs ≥ 118815 would never be generated via greedy decoding.
However, there appears to be space for extra tokens in lm_head, and while the probability is very low, it seems tokens that shouldn't be generated during sampling are occasionally produced.

So I added the logic applying masking logits to ensure next token ID is less then min(model vocab_size, tokenizer's vocab_size)

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

**Comment by ashgold** [comment]
@njhill 
Sorry to bother you. As mentioned in #24211, I created a PR for the issue I registered.
Since it's my first PR, I don't know how to get it merged or who to assign as reviewers, so I just clicked Next and created the PR with the automatically assigned reviewers.

Luckily (?), you were assigned as one of reviewers, so I'm requesting your review. I'm also inquiring whether the PR requires all assigned reviewers to complete their reviews before this PR can be merged.
.
